### ISSUE TITLE
FNX3-16516 ⁃ [Bug] Launching Fenix on adb with app link startup command with URL crashes

### ISSUE THREAD

**Issue by nojunpark** [mozilla-mobile/fenix#2693]
## Steps to reproduce
Connect the phone to ADB
send in this command:
`adb shell am start -W -n org.mozilla.fenix/org.mozilla.fenix.IntentReceiverActivity -a android.intent.action.VIEW -d 'www.mozilla.org'`

### Expected behavior
Fenix opens with the url

### Actual behavior
Crashes with this logcat

```
05-21 13:37:02.937 18016 18016 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{org.mozilla.fenix/org.mozilla.fenix.HomeActivity}: mozilla.appservices.places.PlacesException: Error synchronizing: URL parse error: relative URL without a base
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2861)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2943)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at android.app.ActivityThread.-wrap11(Unknown Source:0)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1630)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:164)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6626)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:811)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: Caused by: mozilla.appservices.places.PlacesException: Error synchronizing: URL parse error: relative URL without a base
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at mozilla.appservices.places.RustError.intoException(RustError.kt:4)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at mozilla.appservices.places.PlacesReaderConnection.getBookmarksWithURL(PlacesConnection.kt:12)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at mozilla.components.browser.storage.sync.PlacesBookmarksStorage$getBookmarksWithUrl$2.invokeSuspend(PlacesBookmarksStorage.kt:3)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:2)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:15)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:1)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:1)
05-21 13:37:02.937 18016 18016 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:42)
05-21 13:37:02.938 18016 18016 I mozac/CrashReporter: Received crash: UncaughtExceptionCrash(throwable=java.lang.RuntimeException: Unable to start activity ComponentInfo{org.mozilla.fenix/org.mozilla.fenix.HomeActivity}: mozilla.appservices.places.PlacesException: Error synchronizing: URL parse error: relative URL without a base)
```

When URL isn't entered, the app does not crash

### Device information
* Android device: Moto G5
* Fenix version: commit `aa9e5c46ae5f4456e866d6c8c82dd9ca9ad74f1f`

**Comment by AaronMT** [comment]
Reproducible on 5ec783f6e5e4f7b9cb7a246228f9f7e35fd7fb01 with an armDebug build on a Pixel 2

**Comment by AaronMT** [comment]
Work-around, supply '**http://**' e.g, 

`$ adb shell am start -W -n org.mozilla.fenix/org.mozilla.fenix.IntentReceiverActivity -a android.intent.action.VIEW -d "http://www.google.com"
`

**Comment by softvision-miralobontiu** [comment]
Verified as fixed on the latest build 1.0.1923 (#11560607), with Xiaomi Mi 8 Lite (Android 8.1).
